flow360/component/simulation/outputs/outputs.py

```diff
@@ -0,0 +1,126 @@
+from typing import List, Optional, Tuple, Union
+
+import pydantic as pd
+
+from flow360.component.simulation.framework.base_model import Flow360BaseModel
+from flow360.component.simulation.framework.entity_base import EntityList
+from flow360.component.simulation.outputs.output_entities import (
+    Isosurface,
+    ProbeMonitor,
+    Slice,
+    SurfaceIntegralMonitor,
+)
+from flow360.component.simulation.primitives import Surface
+from flow360.component.simulation.types import NonNegativeAndNegOneInt
+
+"""Mostly the same as Flow360Param counterparts.
+Caveats:
+1. Check if we support non-average and average output specified at the same time in solver. (Yes but they share the same output_fields)
+2. We do not support mulitple output frequencies for the same type of output.
```

Comment:
how do we enforce this? By validators or by data level design?

Reply:
It is hard to do that from data level design because we allow multiple instances of surfaceOutput inside the output list. They are all independent of each other. I think we will have to do it with validators on the Param level.